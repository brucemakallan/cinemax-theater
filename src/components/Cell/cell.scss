.cell {
  width: 4.76%;
  min-height: 30px;
  padding: 5px;
  display: inline-block;
}
.cellValue {
  background-color: #eaeaea;
  border-radius: 2px;

  button {
    background: transparent;
    border: none;
    cursor: pointer;
    width: 100%;
  }
  button:hover {
    color: green;
  }
}
.vip {
  background-color: rgb(105, 212, 105);
}
.vvip {
  background-color: rgb(123, 182, 238);
}
.twin {
  background-color: rgb(226, 245, 145);
}
.taken {
  background-color: rgb(241, 166, 166);
}
