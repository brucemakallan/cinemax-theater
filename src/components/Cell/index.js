import React from 'react';
import './cell.scss';
import PropTypes from 'prop-types';

const Cell = ({ value, type, id, onClick }) => {
	if (value === 0) value = '';  // don't show the number zero
	return (
		<div className="cell">
			<div className={`cellValue ${type}`}>
				{
					(value === '*')
						? <button type="button" title={type} id={id} onClick={() => onClick(id)}>
							<i className="fas fa-chair"></i>
						</button>
						: value
				}
			</div>
		</div>
	);
};

Cell.propTypes = {
	value: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
};
 
export default Cell;
