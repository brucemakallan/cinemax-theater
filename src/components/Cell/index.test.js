import React from 'react';
import { mount } from 'enzyme';
import Cell from '.';

describe('Test suite for <Cell /> component', () => {
	let props;

	beforeEach(() => {
		props = {
			value: '*',
			type: 'normal',
			id: 'A1',
			onClick: jest.fn()
		};
	});

	it('should render correctly', () => {
		const wrapper = mount(<Cell {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
