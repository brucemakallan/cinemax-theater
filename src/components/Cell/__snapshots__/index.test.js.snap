// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for <Cell /> component should render correctly 1`] = `
<Cell
  id="A1"
  onClick={[MockFunction]}
  type="normal"
  value="*"
>
  <div
    className="cell"
  >
    <div
      className="cellValue normal"
    >
      <button
        id="A1"
        onClick={[Function]}
        title="normal"
        type="button"
      >
        <i
          className="fas fa-chair"
        />
      </button>
    </div>
  </div>
</Cell>
`;
